sources/Context.cpp

```diff
@@ -101,9 +118,115 @@ Context::process() {
   onNodeOnMainConnexComponent(algo::ControllerInterfaceDefinitionAlgorithm(hvdcLines_));
   walkNodesMain();
 
+  // Check all contingencies have their elements present in the network
+  if (def_.simulationKind == SimulationKind::SECURITY_ANALYSIS) {
+    checkContingencies();
+  }
+
   return true;
 }
 
+bool
+Context::isInMainConnectedComponent(const std::string& nodeId) {
+  for (auto& n : mainConnexNodes_) {
+    if (nodeId == n->id) {
+      LOG(debug) << "        node in main connected component " << n->id << LOG_ENDL;
+      return true;
+    }
+  }
+  return false;
+}
+
+bool
+Context::checkGenerator(const std::string& generatorId, std::string& reason) {
+  const auto& network = networkManager_.dataInterface()->getNetwork();
+  for (auto& vl : network->getVoltageLevels()) {
```

Comment:
we were thinking on the best option to use to limit the impact of the change to only DFL: either create indexes for the components of the network interface or for the elements defined in the contingencies. In these alternatives we will have to iterate through all the components of the Network interface at least once.

I assume that extending the Dynawo Network Interface API to allow obtaining a component interface by identifier should not be considered here, right ?

And also, we should discard trying to access the underlying IIDM data (that already contains the required methods to access by id) to avoid dependence on a specific network model.

We were deciding the best way to remove these loops before moving this pull request to "ready for review" (it is still in "draft" mode).